v2/cli/init_command.go

```diff
@@ -310,3 +316,17 @@ func checkFileOverwrite(
 	}
 	return true, false, nil
 }
+
+func validateGitCloneURL(url string) error {
+	gitCloneURLRegex :=
+		regexp.MustCompile(
+			`((git|ssh|http(s)?)|(git@[\w\.]+))` +
+				`(:(//)?)([\w\.@\:/\-~]+)(\.git)(/)?`)
+	urlMatch := gitCloneURLRegex.MatchString(url)
+	if !urlMatch {
+		return fmt.Errorf("invalid value %q for git clone URL"+
+			" (must start with http://, https://, or git@"+
+			" and must end with .git)", url)
+	}
+	return nil
+}
```

Comment:
I can see this being useful in other contexts. Let's move this to `sdk/v2/core` like we did with `ValidateProjectID()`.

Reply:
Which file would this go in? I don't see a specific file in core that fits this function.

Reply:
Let's add a new file called `validations.go`. You can move this _and_ `ValidateProjectID()` to there. Move tests accordingly to `validations_test.go`.